src/lib.rs

```diff
@@ -333,29 +335,45 @@ impl<T: Ord> OrderedCollection<T> {
         X: Ord,
     {
         let x = x.borrow();
-        let mut i = 0;
+        let mut i = 1;
+
         let mask = prefetch_mask(self.items.len());
+        // the search loop is proven to be CPU-backend and not memory bound when using prefetch. So offset part
+        // of prefetch address is intentionally not masked, it allows to do less arithmetic in the loop.
+        // It doesn't affect masking much because `Self::OFFSET` is just half of a cache line.
```

Comment:
Maybe link back to the PR here for future reference?

Reply:
done